lerobot/common/policies/vqbet/modeling_vqbet.py

```diff
@@ -0,0 +1,1173 @@
+            return pred_action, loss
+
+
+class VQBeTHead(nn.Module):
+    def __init__(self, config: VQBeTConfig):
+        """
+        VQBeTHead takes output of GPT layers, and pass the feature through bin prediction head (`self.map_to_cbet_preds_bin`), and offset prediction head (`self.map_to_cbet_preds_offset`)
+
+        self.map_to_cbet_preds_bin: outputs probability of each code (for each layer).
+            The input dimension of `self.map_to_cbet_preds_bin` is same with the output of GPT, 
+            and the output dimension of `self.map_to_cbet_preds_bin` is `self.vqvae_model.vqvae_num_layers (=fixed as 2) * self.config.vqvae_n_embed`.
+            if the agent select the code sequentially, we use self.map_to_cbet_preds_primary_bin and self.map_to_cbet_preds_secondary_bin instead of self._map_to_cbet_preds_bin.
+
+        self.map_to_cbet_preds_offset: output the predicted offsets for all the codes in all the layers. 
+            The input dimension of ` self.map_to_cbet_preds_offset` is same with the output of GPT, 
+            and the output dimension of ` self.map_to_cbet_preds_offset` is `self.vqvae_model.vqvae_num_layers (=fixed as 2) * self.config.vqvae_n_embed * config.action_chunk_size * config.output_shapes["action"][0]`.
+        """
+
+        super().__init__()
+        self.config = config
+        # init vqvae
+        self.vqvae_model = VqVae(config)
+        if config.sequentially_select:
+            self.map_to_cbet_preds_primary_bin = MLP(
+                in_channels=config.gpt_output_dim,
+                hidden_channels=[self.config.vqvae_n_embed],
+            )
+            self.map_to_cbet_preds_secondary_bin = MLP(
+                in_channels=config.gpt_output_dim + self.config.vqvae_n_embed,
+                hidden_channels=[self.config.vqvae_n_embed],
+            )
+        else:
+            self.map_to_cbet_preds_bin = MLP(
+                in_channels=config.gpt_output_dim,
+                hidden_channels=[self.vqvae_model.vqvae_num_layers * self.config.vqvae_n_embed],
+            )
+        self.map_to_cbet_preds_offset = MLP(
+            in_channels=config.gpt_output_dim,
+            hidden_channels=[
+                self.vqvae_model.vqvae_num_layers * self.config.vqvae_n_embed * config.action_chunk_size * config.output_shapes["action"][0],
+            ],
+        )
+        # loss
+        self._focal_loss_fn = FocalLoss(gamma=2.0)
+
+    def discretize(self, n_vqvae_training_steps, actions):
+        if self.vqvae_model.config.action_chunk_size == 1:
+            # not using action chunk
+            actions = actions.reshape(-1, 1, actions.shape[-1])
+        else:
+            # using action chunk
+            slices = []
+            slices.extend([actions[:, j:j+self.vqvae_model.config.action_chunk_size, :] for j in range(actions.shape[1]+1-self.vqvae_model.config.action_chunk_size)])
+            actions = torch.cat(slices, dim=0)
```

Comment:
Could we simplify this?

1. We don't need to make an empty slices than extend it. with a list. We could have just created the list!
2. (nit) We could avoid polluting the namespace with `slices` and just do the list-comprehension in `torch.cat`. This is a style choice I'll leave to you. It depends on how easy it is to read once you format it.
3. As I mentioned above, we can use `self.config` instead of `self.vqvae_model.config`, right?

Reply:
I removed `slices` and just created it using list. Does it seems better?

```
actions = torch.cat(
        [
            actions[:, j : j + self.config.action_chunk_size, :]
            for j in range(actions.shape[1] + 1 - self.config.action_chunk_size)
        ],
        dim=0
    )
```